src/spectra/blackbody.cpp

```diff
@@ -53,6 +61,10 @@ class BlackBodySpectrum final : public Texture<Float, Spectrum> {
 
     BlackBodySpectrum(const Properties &props) : Texture(props) {
         m_temperature = props.float_("temperature");
+        if (props.has_property("scale"))
+            m_scale = props.float_("scale");
+        else
+            m_scale = 1.f;
```

Comment:
You should use `props.float_("scale", 1.f)` here instead of the `if .. else ..` statement